src/util/process_life_cycle.py

```diff
@@ -1,39 +1,293 @@
+                        run_mode=RunMode(self.args.run_mode),
+                        service_fee_calc=self.srvc_fee_calc,
+                        release_override=self.args.release_override,
+                        payment_offset=self.args.payment_offset,
+                        baking_cfg=self.cfg,
+                        life_cycle=self,
+                        payments_queue=self.payments_queue,
+                        dry_run=self.args.dry_run,
+                        client_manager=self.node_client,
+                        node_url=self.args.node_endpoint,
+                        reward_data_provider=self.args.reward_data_provider,
+                        node_url_public=self.args.node_addr_public,
+                        api_base_url=self.args.api_base_url,
+                        retry_injected=self.args.retry_injected).start()
+
+    def do_launch_consumers(self, e):
+        PaymentConsumer(name='consumer0',
+                        payments_dir=self.baking_dirs.payments_root,
+                        key_name=self.cfg.get_payment_address(),
+                        payments_queue=self.payments_queue,
+                        node_addr=self.args.node_endpoint,
+                        client_manager=self.node_client,
+                        plugins_manager=self.plugins_manager,
+                        rewards_type=self.cfg.get_rewards_type(),
+                        args=self.args,
+                        dry_run=self.args.dry_run,
+                        reactivate_zeroed=self.cfg.get_reactivate_zeroed(),
+                        delegator_pays_ra_fee=self.cfg.get_delegator_pays_ra_fee(),
+                        delegator_pays_xfer_fee=self.cfg.get_delegator_pays_xfer_fee(),
+                        dest_map=self.cfg.get_dest_map(),
+                        network_config=self.nw_config,
+                        publish_stats=not self.args.do_not_publish_stats).start()
+
+    def do_shut_down(self, e):
+        logger.info("TRD is shutting down...")
+
         logger.info("--------------------------------------------------------")
         logger.info("Sensitive operations are in progress!")
         logger.info("Please wait while the application is being shut down!")
         logger.info("--------------------------------------------------------")
+
         if self.lock_taken:
-            self.lock_file.release()
+            LockFile().release()
             logger.info("Lock file removed!")
-        self.running = False
-
-    def is_running(self):
-        return self.running
 
     def stop_handler(self, signum, frame):
         logger.info("Application stop handler called: {}".format(signum))
-        self.stop()
+        self.shut_down()
+
+    def shut_down(self):
+        self.fsm.trigger_event(TrdEvent.SHUT_DOWN)
+
+    def is_running(self):
+        return not self.fsm.is_complete()
+
+    def is_dry_run(self, e):
+        return self.args.dry_run
+
+    def is_dry_run_no_consumers(self, e):
```

Comment:
callbacks and condition methods require the "event" argument.

Reply:
If you remove the e argument it will fail? Is this needed by the transistions library ? The examples in the transistions lib seem to be able to use callbacks without any arguments. Or am I missing something?